R/variance_ratio.R

```diff
@@ -33,100 +33,169 @@
+#'  Timothy L. Dickson, Emily C. Farrer, Laureano A. Gherardi, et al. (2014)
+#'  "Biotic Mechanisms of Community Stability Shift along a Precipitation Gradient."
 #'  Ecology 95, no. 6: 1693-1700. doi: 10.1890/13-0895.1
-#'  
-#'  Schluter, Dolph. (1984) "A Variance Test for Detecting Species Associations, 
+#'
+#'  Schluter, Dolph. (1984) "A Variance Test for Detecting Species Associations,
 #'  with Some Example Applications." Ecology 65, no. 3: 998-1005. doi:10.2307/1938071.
 #' @export
-#' @examples 
+#' @examples
 #'  data(knz_001d)
-#'  
+#'
 #'  # Calculate the variance ratio and CIs averaged within replicates
-#'  # Here the null model is repeated once, for final use it is recommended to set a 
+#'  # Here the null model is repeated once, for final use it is recommended to set a
 #'  # large bootnumber (eg, 10000)
-#'  
-#'  res_averagedreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species", 
+#'
+#'  res_averagedreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species",
 #'  abundance.var = "abundance", bootnumber = 1, replicate = "subplot")
-#'  
+#'
 #'  #Calculate the variance ratio and CIs for each replicate
-#'  
-#'  res_withinreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species", 
+#'
+#'  res_withinreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species",
 #'  abundance.var = "abundance", bootnumber = 1, replicate = "subplot", average.replicates = FALSE)
-variance_ratio <- function(df, time.var, species.var, abundance.var, bootnumber, replicate.var=NA,
-                        li=0.025, ui=0.975,  average.replicates=TRUE) {
-  
+variance_ratio <- function(df, time.var, species.var, abundance.var,
+                           bootnumber, replicate.var = NA,
+                           average.replicates = TRUE, level = 0.95, ...) {
+
   # check to make sure abundance is numeric data
   check_numeric(df, time.var, abundance.var)
-   
+
   # if no replicates, calculate single variance ratio
-   if(is.na(replicate.var)) {
-        check_single_onerep(df, time.var, species.var)
-        VR <- variance_ratio_longformdata(df, time.var, species.var, abundance.var)
+  if  (is.na(replicate.var)) {
+
+    ## check the structure of the data
+    check_single_onerep(df, time.var, species.var)
+
+    ## calculate observed variance ratio
+    VR <- variance_ratio_longformdata(df, time.var, species.var, abundance.var)
+
+    ## null models
+
+    nullval <- cyclic_shift(df, time.var = time.var,
+                            species.var = species.var,
+                            abundance.var = abundance.var,
+                            FUN = variance_ratio_matrixdata,
+                            bootnumber = bootnumber)
+
+    nullout <- confint(nullval)
+
+    # bind actual value and CI
+    output <- cbind(nullout, VR)
+
+  } else {
+
+    ## drop any unusued levels
+    df <- droplevels(df)
```

Comment:
I thought this might be a good way to drop unused factor levels (this is from base R). It seems to work when `replicate.var` is a character too